ls1intum/Artemis
# `Programming exercises`: Improve feedback counter in result details

### Checklist
#### General


#### Client

### Motivation and Context


Before this PR, each result detail page would always show the number of tests, sca feedback and manual feedback in total, regardless whether the student clicked on the result on the top right to see all feedbacks or a specific task in the problem statement that only shows selected feedbacks for that task.

### Description

This PR fixes the behavior such that the number above the feedback list only reflects the feedbacks that are actually presented.

Additionally this PR localizes the ResultDetailsView in order to support German as well.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with SCA enabled and manual feedback

1. Log in to Artemis
2. Navigate to Course Administration
3. Participate in the exercise and pass some tests and get some code issues
4. Assess the exercise
5. Check that the result details always show the correct number of passed tests, sea feedback and manual feedback when clicking on the result on the top right as well as any task in the problem statement

### Review Progress






#### Code Review
#### Manual Tests

### Screenshots


Showing all tests, which is wrong:
![Bildschirm­foto 2022-11-03 um 18 44 51](https://user-images.githubusercontent.com/38322605/199796139-15c6bc1f-1a6a-4cdf-92ad-2714f8691f43.png)
Only showing the tests relevant for the current view:
![Bildschirm­foto 2022-11-03 um 18 43 20](https://user-images.githubusercontent.com/38322605/199796207-ee8564b6-436c-45f2-873f-13bfa060c75d.png)
